# Fys Oblig 1

Choose to analyze a curve that looks like:
![a1](attach/realCurveDrawing.png)
with a charge density $\rho_l$

Now the electric field $\vec{E}$ from a small part $\vec{dr}$ would be
![a2](attach/rDrDef.png)

We can see that it's complicated to sum up all the dr's so instead we choose a simplified model

![a3](attach/fakeCurveDrawing.png)

We are going to see how $\vec{E}$ from this model behaves at (0, 0), (0, 5), where it is zero and how all this compares to the simulation of the actual model. It looks like that E at (0,0) wouldnt give an accurate picture of the real model since the charge distribution is moved up, while E at (0, 5) should be much closer to the real deal.

First, the different $\vec{E_q}$'s

Here we are only looking at the xy-plane and we dont care about the x component of $\vec{E}$ because it is obvious that they cancel each other out.
![a4](attach/differentEqs.png)

![a1](attach/Eatzerozero.png)

And at (0, 5)
![a1](attach/Eatzerofive.png)

Now for the simulation. To make them comparable we are going to say that the total charge of the curve is 3q.

In [4]:
import numpy as np
import matplotlib.pyplot as pl


def getE(Qr, R):

    #Qr: the loc of all the Qs
    #R: target loc

    K = 1 #K = 1/4piE_0, just to make it simple, we only care about relative values

    #total charge should be 3q (or rather just 3, since we are setting q=1 for simplicitys sake)
    #so that we can compare with hand calcs
    q = 3/len(Qr)

    E = np.zeros_like(R)
    for i in range(len(Qr)):
        #E from Qi
        r = R - Qr[i]
        E += K*q*r/np.linalg.norm(r)**3
    return E

#place Qs
N=100
x = np.linspace(-2, 2, N)
y = -x**2 + 4

Q = np.array( [ [x[i], y[i] ] for i in range(len(x))] )

#hand calculations gave ca 1.19y for E at (0, 5)
print (f"E(0, 5): {getE(Q, (0., 5.))}")

#and -0.420 at (0,0)
print (f"E(0, 0): {getE(Q, (0., 0.))}")


E(0, 5): [3.01950305e-17 9.48993784e-01]
E(0, 0): [-6.50521303e-17 -2.71104626e-01]


As expected the E at (0,5) is much more accurate ( 1.19 vs 0.949 ) and we can see that since the charge distribution is moved up the vector is greater than it should be.
while E at (0, 0) is relatively way off ( -0.420 vs -0.271 ). Here the vector is also greater even though the charges are farther away from the point than the actual curve, but thats because the E from the curve would point at (0, 0) more from the side, it would be longer in the x direction than y, and of course cancel itself out.

Now lets find some points where E = 0, with brute force ofc. First the simplified model.

In [23]:

#check zero points of E

Q = np.array([
    [-1, 2],
    [0, 4],
    [1, 2]
])

def getEs(Qr, target):

    E = np.zeros_like(target)
    for q in Qr:
        r = target - q
        E += r/np.linalg.norm(r)**3
    
    return E

N = 500

xy = np.array( [ [0, i] for i in np.linspace(0, 5, N)])

E = np.array( [ getEs(Q, xy[i]) for i in range(len(xy)) ])

#all the E = 0

zeroP = []

for i in range(len(E)):
    # if the test number (1e-2) was any smaller the
    # algorith wouldnt spit anything out
    # at N = 10000 1e-3 does work, while 1e-4 gives nothing
    if (np.linalg.norm(E[i]) < 1e-2):
        zeroP.append(xy[i])

print (zeroP)

[array([0.        , 2.15430862]), array([0.        , 2.84569138])]


So around y=2.15 and y=2.84 E is basically zero. Makes sense from looking at the drawing that it would have two zero points, two balance points between (Eq1 and Eq3) and Eq2. One where theyre both smaller(closer to q1 and q3) and one where theyre both bigger(closer to q2). These are the same points that wolfram alpha gave when i gave it the actual equations. It seems that N has to be ridiculously large for E to approach say 1e-16.

Now the actual field

In [20]:

N=600
x = np.linspace(-2, 2, N)
y = -x**2 + 4

Q = np.array( [ [x[i], y[i] ] for i in range(len(x))] )

R = np.array( [ [0., i] for i in np.linspace(0, 4, N) ])

allE = [ getE(Q, r) for r in R ]

zeroP = []

for i in range(len(allE)):

    if (np.linalg.norm(allE[i]) < 1e-1):
        zeroP.append(xy[i])

print (zeroP)


[]


Well... Maybe it changes direction too fast and we dont have enough N's to find the 0. It is obvious that the field points downward when we start at (0,0), so lets do a test to see if it ever gets positive in the y direction, if so then there must a zero point since it is continuous.

In [30]:
pointsCount = 0
alreadyPos = False
for i in range(len(allE)):

    if (allE[i][1] > 0 and i<(len(allE)-1)): #exclude the last obvious case
        if (not alreadyPos):
            pointsCount += 1
            alreadyPos = True
    else:
        alreadyPos = False
            
print(pointsCount)



0


So there probably arent any zero points. And this would be the greatest difference between the simplified model and the real one. Kinda a big one.